motionplan/constraint.go

```diff
@@ -493,3 +493,154 @@ func NewOctreeCollisionConstraint(octree *pointcloud.BasicOctree, threshold int,
 	}
 	return constraint
 }
+
+// ------------ NEW RDK BASED CODE ------------
+// LinearConstraint specifies that the component being moved should move linearly relative to its goal.
+// It does not constrain the motion of components other than the `component_name` specified in motion.Move
+type LinearConstraint struct {
+	LineToleranceMm          *float32 // Max linear deviation from straight-line between start and goal, in mm.
+	OrientationToleranceDegs *float32
+}
+
+// OrientationConstraint specifies that the component being moved will not deviate its orientation beyond some threshold relative
+// to the goal. It does not constrain the motion of components other than the `component_name` specified in motion.Move
+type OrientationConstraint struct {
+	OrientationToleranceDegs *float32
+}
+
+type CollisionSpecification_AllowedFrameCollisions struct {
+	Frame1 string
+	Frame2 string
+}
+
+// CollisionSpecification is used to selectively apply obstacle avoidance to specific parts of the robot
+type CollisionSpecification struct {
+	// Pairs of frame which should be allowed to collide with one another
+	Allows []*CollisionSpecification_AllowedFrameCollisions
+}
+
+// Constraints is a struct to store the constraints imposed upon a robot
+// It serves as a convenenient RDK wrapper for the protobuf object
+type Constraints struct {
+	LinearConstraint       []*LinearConstraint
+	OrientationConstraint  []*OrientationConstraint
+	CollisionSpecification []*CollisionSpecification
```

Comment:
I think lets remove the pointers from these types as well. Since the structs are pretty small we dont mind passing by value